scripts/apps/search/components/item-list-popup.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {ElementType, CSSProperties, SyntheticEvent} from 'react';
+import {createPortal} from 'react-dom';
+import {positionPopup} from '../helpers';
+
+interface IProps {
+    label: string;
+    close: () => void;
+    target: ElementType | EventTarget;
```

Comment:
I use to avoid storing refs in translations component

Reply:
I'm suggesting to change the type from `target: ElementType | EventTarget;` to `target: HTMLElement;`. It shouldn't have to do anything with refs, you would still store it in the state as you do now.

Reply:
wouldn't that re-render just to store those refs?

Reply:
If you used refs, it would re-render. It doesn't matter if it re-renders since it's a fraction of a second anyway. That said, I don't suggest to use refs.

```
popup: {
    label: gettext('Translations'),
    ids: this.props.item.translations,
    target: elem, // <-- `HTMLElement`
},
```

I've tracked the usages and the `elem` you use in the current `setState` call is already `HTMLElement` it's just that TypeScript can't guarantee it, in which case it's fine to use type casting and do ` target: elem as HTMLElement`.